crates/ecolor/src/color32.rs

```diff
@@ -95,21 +105,17 @@ impl Color32 {
     /// From `sRGBA` WITHOUT premultiplied alpha.
     #[inline]
     pub fn from_rgba_unmultiplied(r: u8, g: u8, b: u8, a: u8) -> Self {
-        if a == 255 {
-            Self::from_rgb(r, g, b) // common-case optimization
-        } else if a == 0 {
-            Self::TRANSPARENT // common-case optimization
-        } else {
-            let r_lin = linear_f32_from_gamma_u8(r);
-            let g_lin = linear_f32_from_gamma_u8(g);
-            let b_lin = linear_f32_from_gamma_u8(b);
-            let a_lin = linear_f32_from_linear_u8(a);
-
-            let r = gamma_u8_from_linear_f32(r_lin * a_lin);
-            let g = gamma_u8_from_linear_f32(g_lin * a_lin);
-            let b = gamma_u8_from_linear_f32(b_lin * a_lin);
-
-            Self::from_rgba_premultiplied(r, g, b, a)
+        match a {
```

Comment:
Might be worth checking if skipping the match for the 0/255 cases is actually faster. At least on my machine, it is.

With match
```
from_rgba_unmultiplied_100
                        time:   [2.0122 ns 2.0443 ns 2.0813 ns]
```

Just using the table
```
from_rgba_unmultiplied_100
                        time:   [1.6584 ns 1.6675 ns 1.6772 ns]
                        change: [-18.696% -17.776% -16.940%] (p = 0.00 < 0.05)
                        Performance has improved.
```

Reply:
Yes, but the common cases become slower, so I decided to just leave it as it was before.

Reply:
Yeah, that is true.